src/applications/bmqstoragetool/m_bmqstoragetool_searchresult.h

```diff
@@ -577,6 +688,96 @@ class SearchGuidDecorator : public SearchResultDecorator {
     void outputResult() BSLS_KEYWORD_OVERRIDE;
 };
 
+// ===========================
+// class SearchOffsetDecorator
+// ===========================
+
+/// This class provides decorator to handle search of given offsets.
+class SearchOffsetDecorator : public SearchResultDecorator {
+  private:
+    // PRIVATE DATA
+    bsl::vector<bsls::Types::Int64> d_offsets;
+    // List of offsets to search for.
+    bsl::ostream& d_ostream;
+    // Reference to output stream.
+    bool d_withDetails;
+    // If 'true', output detailed result, output short one otherwise.
+
+  public:
+    // CREATORS
+
+    /// Constructor using the specified `component`, `offsets`, `ostream`,
+    /// `withDetails` and `allocator`.
+    SearchOffsetDecorator(const bsl::shared_ptr<SearchResult>&   component,
+                          const bsl::vector<bsls::Types::Int64>& offsets,
+                          bsl::ostream&                          ostream,
+                          bool                                   withDetails,
+                          bslma::Allocator*                      allocator);
+
+    // MANIPULATORS
+
+    /// Process `message` record with the specified `record`, `recordIndex` and
+    /// `recordOffset`.
+    bool processMessageRecord(const mqbs::MessageRecord& record,
+                              bsls::Types::Uint64        recordIndex,
+                              bsls::Types::Uint64        recordOffset)
+        BSLS_KEYWORD_OVERRIDE;
+    /// Process `deletion` record with the specified `record`, `recordIndex`
+    /// and `recordOffset`.
+    bool processDeletionRecord(const mqbs::DeletionRecord& record,
+                               bsls::Types::Uint64         recordIndex,
+                               bsls::Types::Uint64         recordOffset)
+        BSLS_KEYWORD_OVERRIDE;
+    /// Output result of a search.
+    void outputResult() BSLS_KEYWORD_OVERRIDE;
+};
+
+// ===================================
+// class SearchSequenceNumberDecorator
+// ===================================
+
+/// This class provides decorator to handle search of given composite sequence
+/// numbers.
+class SearchSequenceNumberDecorator : public SearchResultDecorator {
+  private:
+    // PRIVATE DATA
+    bsl::vector<CompositeSequenceNumber> d_seqNums;
+    // List of composite sequence numbers to search for.
```

Comment:
Most of the code uses this BDE-style format docs for data members, but it's because the conversion script did not catch them.  I'm slowly converting this over, but it's a manual process (package by package in https://github.com/bloomberg/blazingmq/pull/523).  For new code, let's use doxygen style for everything (`///` before the members and Markdown).

Not a big deal, but less to change later.